File: react/numberdatasource.js
//#pragma: global
    let datasource[[id]] = new class {
    #listeners
    #value;
    constructor(initialvalue){  
        this.#listeners = []
        this.#value = initialvalue;
    }

    refresh(v){ this.#value = this.#value+1; this.notifyListeners(); }
    setValue(v){ this.#value = v; this.notifyListeners(); }
    getValue() { return this.#value }
    
    addListener(l){
        this.#listeners.push(l)
    }
    notifyListeners(){
        this.#listeners.forEach(l => {
            // console.log("l", l); 
            l(this.#value);
        })
    }
}(1);

//#pragma: setup
 datasource[[id]].addListener(() => this.setState({}))
    

File: flutter/numberdatasource.dart
//#pragma: global
   NumberDataSource datasource[[id]] = NumberDataSource();

//#pragma: setup
        datasource[[id]].addListener((){
          setState(() {});
        });

//#pragma: declaration
class NumberDataSource extends ChangeNotifier {
  int _value =  0;
  int get value => _value;
  set value(v) {
      _value = v;
      notifyListeners();
    
  }

  _refresh(e) {
    _value = _value + 1;
  }

  refresh(e) {
    _refresh(e); 
    notifyListeners(); 
  }

  int getValue() => _value; 
  setValue(v) { _value = v; notifyListeners(); }

  }
         